packages/page-claims/src/index.tsx

```diff
@@ -73,180 +67,194 @@ const Signature = styled.textarea`
   }
 `;
 
-// FIXME Not React.FC yet
-class ClaimsApp extends TxModal<Props, State> {
-  constructor (props: Props) {
-    super(props);
-
-    this.defaultState = {
-      ...this.defaultState,
-      claim: null,
-      didCopy: false,
-      ethereumAddress: null,
-      signature: null,
-      step: 0
-    };
-    this.state = this.defaultState;
-  }
+const ClaimsApp = (props: Props): React.ReactElement => {
+  const [isOldClaimProcess, setIsOldClaimProcess] = useState(true);
+  const [claim, setClaim] = useState<Balance | null>(null);
+  const [didCopy, setDidCopy] = useState(false);
+  const [ethereumAddress, setEthereumAddress] = useState<EthereumAddress | null>(null);
+  const [signature, setSignature] = useState<EcdsaSignature | null>(null);
+  const [step, setStep] = useState<Step>(Step.Account);
+  const [accountId, setAccountId] = useState<string | null>(null);
+  const { api, isApiReady, systemChain } = useApi();
+  const { t } = useTranslation();
+  const [isPreclaimed, setIsPreclaimed] = useState(false);
+
+  useEffect(() => {
+    if (isApiReady && typeof api.query.claims.claimAttest !== undefined) {
+      setIsOldClaimProcess(false);
```

Comment:
Is there a guarantee that isApiReady will be `true` before rendering this component? Like somewhere higher in the components tree?

If yes, we could simplify this useEffect&useState into just
```ts
const isOldClaimProcess = !!api.query.claims.claimAttest;
```

Reply:
Indeed. As long as the route has `needsApi` set - https://github.com/polkadot-js/apps/blob/master/packages/apps/src/Content/index.tsx#L50

Reply:
... which it does - https://github.com/polkadot-js/apps/blob/master/packages/apps-routing/src/claims.ts#L13

Reply:
The reason to never use the promises directly also means the boilerplate is reduced and you also don't need to resort to `useMountedRef` (so unmounts won't spam the console, state updates are only done _while_ mounted)